MeshLib/MeshEditing/AddLayerToMesh.cpp

```diff
@@ -114,7 +115,15 @@ MeshLib::Mesh* addLayerToMesh(MeshLib::Mesh const& mesh, double thickness,
 			extrudeElement(subsfc_nodes, *sfc_elements[k], subsfc_sfc_id_map)
 		);
 
-	MeshLib::Properties subsfc_props (mesh.getProperties());
+	MeshLib::Properties subsfc_props(mesh.getProperties());
+	// remove all but the MaterialIDs properties
+	std::vector<std::string> prop_names(subsfc_props.getPropertyVectorNames());
+	for (auto const& name : prop_names) {
+		if (name != "MaterialIDs") {
+			subsfc_props.removePropertyVector(name);
+		}
+	}
+	// update the MaterialIDs property
 	boost::optional<MeshLib::PropertyVector<int> &> opt_materials(
 		subsfc_props.getPropertyVector<int>("MaterialIDs")
 	);
```

Comment:
If you only leave MatIDs in there, wouldn't it be faster/better to create a new Properties object instead of first copying everything and then deleting most of it?